### PR TITLE
Extend id token duration if tests included as part of pipeline

### PR DESCRIPTION
**Pull request by noah-paige**
### Feature or Bugfix

- Feature

### Detail
- Extend ID Token Validity to 120 if enabling integration tests to allow integration test suite to run  fully (note: runs take ~70-80 minutes currently) 

### Relates
- NA

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: deploy/stacks/cognito.py [hunk 1]
```diff
@@ -100,13 +100,15 @@ def __init__(
                 domain_prefix=f"{resource_prefix.replace('-', '')}{envname}{self.region.replace('-', '')}{self.account}"
             ),
         )
+        id_token_duration = 120 if with_approval_tests else 60
```
Comment (dlpzx) [inline deploy/stacks/cognito.py:103]: The default is 30, should we keep the `with_approval_test` False =30?

**Review by dlpzx — APPROVED** [review]
Left one minor comment

File: deploy/stacks/cognito.py (hunk 1 above)
Comment (noah-paige) [reply deploy/stacks/cognito.py:103]: I thought it was 60 minutes - https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_cognito/UserPoolClientProps.html

Comment (dlpzx) [reply deploy/stacks/cognito.py:103]: For some reason in my deployments it was set as 30, all good

**Pull request merged**